core/src/main/java/org/mapstruct/DecoratedWith.java

```diff
@@ -145,7 +145,7 @@
  * @author Gunnar Morling
  */
 @Target(ElementType.TYPE)
-@Retention(RetentionPolicy.SOURCE)
+@Retention(RetentionPolicy.CLASS)
```

Comment:
I see this is one of the last ones that does not have the retention policy 'CLASS'.  Only 'Experimental'  remaining.

I guess a more important reason is that some annotations need to be referenced in libraries. Although, that's probably not the case for `@DecoratedWith` but also not for `@Mapper`. I forgot why runtime was not sufficient for the latter one, but I guess`@DecoratedWith` needs to be in line with `@Mapper`

Reply:
`Experimental` is indeed only `SOURCE`, but that one is only for documentation, so that is fine.

`@Mapper` might actually be used in other libraries, e.g. if you use other mappers via `Mapper#uses`. `RUNTIME` is not needed since we don't want our annotations to be present during the runtime execution.